python/spark_esri/__init__.py

```
@@ -116,20 +343,44 @@ def spark_start(config: Dict = {}) -> SparkSession:
+            SparkContext._active_spark_context.stop()
+    finally:
+        proc = getattr(gateway, "proc", None)
+        try:
+            gateway.shutdown()
+        except Exception:  # noqa: BLE001 - the JVM may already be gone
+            pass
+        if proc is not None:
+            if proc.stdin is not None:
+                proc.stdin.close()
+            # ensure that process and all children are killed
+            subprocess.Popen(["cmd", "/c", "taskkill", "/f", "/t", "/pid", str(proc.pid)],
```

Ruff (0.16.0)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)